cockatrice/src/client/network/parsers/deck_link_to_api_transformer.cpp

```diff
@@ -0,0 +1,61 @@
+#include "deck_link_to_api_transformer.h"
+
+#include <QRegularExpression>
+
+namespace DeckLinkToApiTransformer
+{
+
+static const QString TAPPEDOUT_BASE = "https://tappedout.net/mtg-decks/";
+static const QString TAPPEDOUT_SUFFIX = "/?fmt=txt";
+
+static const QString ARCHIDEKT_BASE = "https://archidekt.com/api/decks/";
+static const QString ARCHIDEKT_SUFFIX = "/?format=json";
+
+static const QString MOXFIELD_BASE = "https://api.moxfield.com/v2/decks/all/";
+static const QString MOXFIELD_SUFFIX = "/";
+
+static const QString DECKSTATS_SUFFIX = "?include_comments=1&export_mtgarena=1";
+
+bool parseDeckUrl(const QString &url, ParsedDeckInfo &outInfo)
+{
+    QRegularExpression rxTappedOut("tappedout\\.net/(?:mtg-decks/)?([^/?#]+)");
+    QRegularExpression rxArchidekt("archidekt\\.com/decks/(\\d+)");
+    QRegularExpression rxMoxfield("moxfield\\.com/decks/([a-zA-Z0-9_-]+)");
+    QRegularExpression rxDeckstats("deckstats\\.net/decks/(\\d+/[a-zA-Z0-9_-]+)");
```

Comment:
I don't know if the compiler optimizes out recompiling the regex on every call. I make all regex variables static just to be safe
```suggestion
    static QRegularExpression rxTappedOut("tappedout\\.net/(?:mtg-decks/)?([^/?#]+)");
    static QRegularExpression rxArchidekt("archidekt\\.com/decks/(\\d+)");
    static QRegularExpression rxMoxfield("moxfield\\.com/decks/([a-zA-Z0-9_-]+)");
    static QRegularExpression rxDeckstats("deckstats\\.net/decks/(\\d+/[a-zA-Z0-9_-]+)");
```